tests/sentry/api/endpoints/test_sentry_app_details.py

```diff
@@ -219,12 +267,185 @@ def test_cannot_update_scopes_published_app(self):
         response = self.get_error_response(
             self.published_app.slug,
             name="NewName",
-            webookUrl="https://newurl.com",
+            webhookUrl="https://newurl.com",
             scopes=("project:read",),
             status_code=400,
         )
         assert response.data["detail"] == "Cannot update permissions on a published integration."
 
+    def test_add_service_hooks_and_update_scope(self):
+        # first install the app on two organizations
+        org1 = self.create_organization(name="Org1")
+        org2 = self.create_organization(name="Org2")
+
+        installation1 = self.create_sentry_app_installation(
+            organization=org1, slug=self.published_app.slug
+        )
+        installation2 = self.create_sentry_app_installation(
+            organization=org2, slug=self.published_app.slug
+        )
+
+        assert installation1.organization_id == org1.id
+        assert installation2.organization_id == org2.id
+        assert installation1.sentry_app == self.published_app
+        assert installation2.sentry_app == self.published_app
+        self.published_app.scope_list = ("event:write", "event:read")
+        self.published_app.save()
+
+        # for published integrations, it runs in a task
+        print(
+            "start test with organization ids: ",
+            org1.id,
+            org2.id,
+            "and application id: ",
+            self.published_app.application_id,
+        )
+        with self.tasks(), outbox_runner():
+            self.get_success_response(
+                self.published_app.slug,
+                webhookUrl="https://newurl.com",
+                scopes=("event:read", "event:write"),
+                events=("issue",),
+                status_code=200,
+            )
+        print(
+            f"task and outbox runner should be done by now with organization ids: {org1.id}, {org2.id}, application id: {self.published_app.application_id}"
+        )
+        self.published_app.refresh_from_db()
+        assert set(self.published_app.scope_list) == {"event:write", "event:read"}
+        assert (
+            self.published_app.webhook_url == "https://newurl.com"
+        ), f"Unexpected webhook URL: {self.published_app.webhook_url}"
+        # Check service hooks for each organization
+        with assume_test_silo_mode(SiloMode.REGION):
+            service_hooks_org1 = ServiceHook.objects.filter(
+                organization_id=org1.id, application_id=self.published_app.id
+            )
+            service_hooks_org2 = ServiceHook.objects.filter(
+                organization_id=org2.id, application_id=self.published_app.id
+            )
```

Comment:
Shouldn't these conditions be `self.published_app.application_id` ?

Reply:
@markstory You're a genius! And that was the problem. I guess what happened is that in my local env, the ids just happened to be the same so I didn't catch it. I bet there are a lot of bugs caused by that sort of behavior not being caught in tests.